### PR TITLE
adds authentication feature to routes

### PR DESCRIPTION
**Pull request by neriyahbutler**
Previous PR was messed up because I made a mistake when trying to revert commits I accidentally added. Also, comments left on the previous PR were resolved in this one

### CODE REVIEW

File: functions/src/errorChecks.ts [hunk 1]
```diff
@@ -1,69 +1,69 @@
 export function checkCVEValidity(ID: any): boolean {
```
Comment (arjun-garg-google) [inline functions/src/errorChecks.ts:1]: what's with the formatting changes in this file?

Comment (neriyahbutler) [reply functions/src/errorChecks.ts:1]: I can fix it in a different PR

**Comment by neriyahbutler** [comment]
Made changes to match Helena's removal of express and changed all the 'POST's to 'GET's

File: functions/src/index.ts [hunk 2]
```diff
@@ -14,20 +15,22 @@ export const getBulletinFunction = bulletinFunction.getBulletin;
...
 
-export const getData = functions.https.onRequest(main);
-
-app.post('/data', (request: any, response: any) => {
-  if (request.body['email']) {
-    const email: string = request.body['email'];
-    setAdminPriveleges(email).catch(error => {
-        response.status(400).send("error giving admin privileges:"+ error);
-      }
+export const grantAdminRole = functions.https.onRequest((request: any, response: any) => {
```
Comment (arjun-garg-google) [inline functions/src/index.ts:18]: separate this out into its own files like for the other cloud functions

**Pull request merged**